### PR TITLE
Review/swift3.0 [no merge]

### LINKED ISSUE #111: Swift 3.0 support
- [x] Update pod, example and playground to swift 
- [x] Update README code
- [x] Add `.swift_version` file https://github.com/CocoaPods/CocoaPods/pull/5841
- [x] Create branch swift 2.3 from last commit before merging
- [ ] Release to cocoa pods as 2.0.0

### PR DESCRIPTION
**Pull request by joanromano**
Closes #111

### CODE REVIEW

**Comment by MP0w** [comment]
FYI @joanromano I created `review/swift3Base` and changed the destination of this PR so we can check the diffs... otherwise the changes in the pods were too many

**Comment by MP0w** [comment]
-JSONAPILink seems broken and tests are hanging... @joanromano any idea?-
fixed, was a SwiftyJSON BCH

**Review by MP0w — APPROVED** [review]
Approved from my side, once @joanromano approves the other PRs and this one the steps are:
1. Merge README and Example branches into this one
2. Change destination branch of PR to master
3. Create swift/2.3 branch from master
4. Merge
5. Release 2.0.0

File: Tests/RouterTests.swift [hunk 1]
```diff
@@ -453,14 +451,14 @@ class RouterTests: QuickSpec {
...
+                var responseDictionary: [String: AnyObject]?
                 
                 router.get("/users/:id") { request in
                     return store.find(User.self, id: request.components["id"]!)
                 }
                 
-                NSURLSession.sharedSession().dataTaskWithURL(NSURL(string: "http://www.test.com/users/1")!) { (data, response, _) in
-                    responseDictionary = try! NSJSONSerialization.JSONObjectWithData(data!, options: .MutableLeaves) as? NSDictionary
+                URLSession.shared.dataTask(with: URL(string: "http://www.test.com/users/1")!) { (data, response, _) in
+                    responseDictionary = try! JSONSerialization.jsonObject(with: data!, options: .mutableLeaves) as? Dictionary
```
Comment (joanromano) [inline Tests/RouterTests.swift:461]: Change to `[String: AnyObject]?`

**Review by joanromano — COMMENTED** [review]
We should also consider some method renamings to adopt new Swift guidelines.

**Pull request closed without merging**